rakelib/artifacts.rake

```diff
@@ -669,7 +669,7 @@ namespace "artifact" do
       elsif stat.symlink?
         tar.symlink(path_in_tar, File.readlink(path), :mode => stat.mode)
       else
-        tar.add_file_simple(path_in_tar, :mode => stat.mode, :size => stat.size) do |io|
+        tar.add_file_simple(path_in_tar, :mode => stat.mode, :size => stat.size, :mtime => stat.mtime&.to_i || 0) do |io|
```

Comment:
For the reviewer here is the source for this method https://github.com/halostatue/minitar/blob/a531136b17b9efdddf0a0f39537845b454c2371e/lib/minitar/writer.rb#L104-L177

Idea is that to avoid it defaulting to nil (and resulting in a 0 for linux epoch time) we explicitly pass through the mtime of the file on the builder (or 0 if for some reason that cant be computed). 

One use case https://github.com/elastic/logstash/issues/17925 on upgrade is relying on rsync defaults where it would be convenient that for each logstash release *all* files in an artifact have a "newer" mtime than the previous release. I believe this would ensure that.

Reply:
Should we then just do `:mtime => stat.mtime&.to_i || Time.now.to_i` ? seems better than epoch, there's already some expectation that dates of files in the archive will be similar to the date of when the files were put into the archive.

Reply:
Yeah, i can do that. I highly doubt `stat.mtime` would ever *not* give us a time, but i think using `Time.now` logically makes more sense than 0.